src/main/java/com/direwolf20/buildinggadgets/common/items/capability/GadgetCapabilityProvider.java

```diff
@@ -0,0 +1,60 @@
+package com.direwolf20.buildinggadgets.common.items.capability;
+
+import com.direwolf20.buildinggadgets.common.building.CapabilityBlockProvider;
+import com.direwolf20.buildinggadgets.common.building.IBlockProvider;
+import com.direwolf20.buildinggadgets.common.config.SyncedConfig;
+import com.direwolf20.buildinggadgets.common.tools.GadgetUtils;
+import com.google.common.base.Preconditions;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumFacing;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.capabilities.ICapabilityProvider;
+import net.minecraftforge.energy.CapabilityEnergy;
+import net.minecraftforge.energy.IEnergyStorage;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.function.IntSupplier;
+
+public class GadgetCapabilityProvider implements ICapabilityProvider {
```

Comment:
https://github.com/Direwolf20-MC/BuildingGadgets/pull/240#discussion_r260175161
If this is not clear enough, I will state it one more time:
Minecraft/Forge only supports **1** capability provider object to be given to an item at a time. If I wasn't to rename it, it would be a capability provider naed 'CapabilityProviderEnergy' that does both energy capability and block provider capability, which is very confusing and improperly formatted code.

Reply:
To state it one more time: it is a matter of a few lines to provide a compound Wrapper implementation and definitly not out of Scope for this PR...